pkg/pod/pod.go

```diff
@@ -384,3 +384,13 @@ func shouldAddReadyAnnotationOnPodCreate(ctx context.Context, sidecars []v1beta1
 	cfg := config.FromContextOrDefaults(ctx)
 	return !cfg.FeatureFlags.RunningInEnvWithInjectedSidecars
 }
+
+// WillOverwritePodSetAffinity returns a bool indicating whether the
+// affinity for pods will be overwritten with affinity assistant.
+func WillOverwritePodSetAffinity(taskRun *v1beta1.TaskRun) bool {
+	var podTemplate v1beta1.PodTemplate
+	if taskRun.Spec.PodTemplate != nil {
+		podTemplate = *taskRun.Spec.PodTemplate
+	}
+	return taskRun.Annotations[workspace.AnnotationAffinityAssistantName] != "" && podTemplate.Affinity != nil
```

Comment:
Is it possible that `taskRun.Annotations` is ever a nil map?  I think this might panic if so.

Reply:
Thank you for the review. I did not introduce a nil check here as it is only reading the map so there won't be a panic if the map is nil. Please let me know if I am missing something here regarding the nil check.